cloud/parameterstore/parameter_manager.go

```diff
@@ -0,0 +1,196 @@
+	if pathPrefix != "" {
+		pathPrefix = fmt.Sprintf("/%s/", strings.TrimPrefix(strings.TrimSuffix(pathPrefix, "/"), "/"))
+	}
+	return &ParameterManager{
+		pathPrefix:     pathPrefix,
+		cachingEnabled: cachingEnabled,
+		ssmClient:      ssmClient,
+		db:             db,
+	}
+}
+
+// Put adds or updates a parameter. This returns the created parameter.
+func (pm *ParameterManager) Put(ctx context.Context, name, value string) (*Parameter, error) {
+	fullName := pm.getPrefixedName(name)
+	if _, err := pm.ssmClient.PutParameter(ctx, &ssm.PutParameterInput{
+		Name:      aws.String(fullName),
+		Value:     aws.String(value),
+		Overwrite: aws.Bool(true),
+		Type:      ssmTypes.ParameterTypeSecureString,
+		Tier:      ssmTypes.ParameterTierIntelligentTiering,
+	}); err != nil {
+		return nil, errors.Wrapf(err, "putting parameter '%s'", name)
+	}
+
+	// TODO (DEVPROD-9403): update cache as needed.
+
+	return &Parameter{
+		Name:     fullName,
+		Basename: pm.getBasename(fullName),
+		Value:    value,
+	}, nil
+}
+
+// Get retrieves the parameters given by the provided name(s). If some
+// parameters cannot be found, they will not be returned. Use GetStrict to both
+// get the parameters and validate that all the requested parameters were found.
+func (pm *ParameterManager) Get(ctx context.Context, names ...string) ([]Parameter, error) {
+	if len(names) == 0 {
+		return nil, nil
+	}
+
+	// TODO (DEVPROD-9403): use caching if possible before retrieving value from
+	// Parameter Store.
+
+	fullNames := make([]string, 0, len(names))
+	for _, name := range names {
+		fullNames = append(fullNames, pm.getPrefixedName(name))
+	}
+
+	ssmParams, err := pm.ssmClient.GetParametersSimple(ctx, &ssm.GetParametersInput{
+		Names:          fullNames,
+		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting %d parameters", len(names))
+	}
+
+	params := make([]Parameter, 0, len(ssmParams))
+	for _, p := range ssmParams {
+		name := aws.ToString(p.Name)
+		params = append(params, Parameter{
+			Name:     name,
+			Basename: pm.getBasename(name),
+			Value:    aws.ToString(p.Value),
+		})
+	}
+
+	return params, nil
+}
+
+// GetStrict is the same as Get but verifies that all the requested parameter
+// names were found before returning the result.
+func (pm *ParameterManager) GetStrict(ctx context.Context, names ...string) ([]Parameter, error) {
+	fullNames := make([]string, 0, len(names))
+	for _, name := range names {
+		fullNames = append(fullNames, pm.getPrefixedName(name))
+	}
+
+	params, err := pm.Get(ctx, names...)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(params) != len(fullNames) {
+		foundNames := make(map[string]struct{}, len(params))
+		for _, p := range params {
+			foundNames[p.Name] = struct{}{}
+		}
+
+		var missingNames []string
+		for _, name := range fullNames {
+			if _, ok := foundNames[name]; !ok {
+				missingNames = append(missingNames, name)
```

Comment:
Since we don't return the results on error, can we just return here? Or is it important to have the full list of missing parameters?

Reply:
It's more useful to have a full list. Since this is just iterating through a list (one that can't be very long in practice), it's easy and more complete to just list out all the invalid parameters rather than give an incomplete error message.